engine/gameobject/src/gameobject/gameobject.cpp

```diff
@@ -966,9 +966,11 @@ namespace dmGameObject
         return instance;
     }
 
-    dmhash_t ConstructInstanceId(uint32_t index)
+    dmhash_t ConstructInstanceId()
     {
-        char buffer[16] = { 0 };
+        static uint32_t index = 0;
+        index += 1;
+        char buffer[32] = { 0 };
```

Comment:
We need at least 20 bytes `/` + `instance` + `uint32` + = 1 + 8 + 10 + 1

Reply:
Perhaps a code comment?